pysmt/test/test_optimization.py

```diff
@@ -21,45 +21,97 @@
 from pysmt.shortcuts import Optimizer, GE, Int, Symbol, INT, LE, GT, REAL, Real
 from pysmt.shortcuts import And, Plus, Minus, get_env
 from pysmt.logics import QF_LIA, QF_LRA
+from pysmt.optimization.goal import MaximizationGoal, MinimizationGoal, \
+    MinMaxGoal, MaxMinGoal
 
 from pysmt.exceptions import PysmtUnboundedOptimizationError
 
 class TestOptimization(TestCase):
 
     @skipIfNoOptimizerForLogic(QF_LIA)
-    def test_basic(self):
+    def test_minimization_basic(self):
         x = Symbol("x", INT)
+        min = MinimizationGoal(x)
         formula = GE(x, Int(10))
         for oname in get_env().factory.all_optimizers(logic=QF_LIA):
             with Optimizer(name=oname) as opt:
                 opt.add_assertion(formula)
-                model, cost = opt.optimize(x)
+                model, cost = opt.optimize(min)
                 self.assertEqual(model[x], Int(10))
                 self.assertEqual(cost, Int(10))
 
+
+    @skipIfNoOptimizerForLogic(QF_LIA)
+    def test_maximization_basic(self):
+        x = Symbol("x", INT)
+        max = MaximizationGoal(x)
+        formula = LE(x, Int(10))
+        for oname in get_env().factory.all_optimizers(logic=QF_LIA):
+            with Optimizer(name=oname) as opt:
+                opt.add_assertion(formula)
+                model, cost = opt.optimize(max)
+                self.assertEqual(model[x], Int(10))
+
+    @skipIfNoOptimizerForLogic(QF_LIA)
+    def test_minmax_basic(self):
+        x = Symbol("x", INT)
+        y = Symbol("y", INT)
+        z = Symbol("z", INT)
+
+        f11 = GE(x, Int(5))
+        f12 = LE(x, Int(8))
+
+        f21 = GE(y, Int(11))
+        f22 = LE(y, Int(14))
+
+        f31 = GE(z, Int(15))
+        f32 = LE(z, Int(18))
+
+        minmax = MinMaxGoal([x, y, z])
+
+        optimizers = ["z3_sua", "z3_incr", "msat_sua", "msat_incr", "optimsat"]
```

Comment:
Nope. If you do like this this test will fail if some of the solvers are not installed. you can do like this:
```
for oname in get_env().factory.all_optimizers(logic=QF_LIA):            
    with Optimizer(name=oname) as opt:
```
you can use an if over `oname` if you want to skip solvers, but here it should not be needed